src/terminal/color.lua

```diff
@@ -0,0 +1,301 @@
+--- Module for setting text colors and attributes.
+-- contains other functionality such as enabling/disabling text attributes like underline, blinking
+-- and reversing. Also contains functionality for setting brightness levels.
+-- @module terminal.color
+-- @usage
+-- local color = require("terminal.color")
+local M = {}
+package.loaded["terminal.color"] = M
+local output = require("terminal.output")
+
+local fg_color_reset = "\27[39m"
+local bg_color_reset = "\27[49m"
+local attribute_reset = "\27[0m"
+local underline_on = "\27[4m"
+local underline_off = "\27[24m"
+local blink_on = "\27[5m"
+local blink_off = "\27[25m"
+local reverse_on = "\27[7m"
+local reverse_off = "\27[27m"
+
+M.fg_base_colors = setmetatable({
```

Comment:
these are internals, should we be exporting them? Do we need them?

if for testing we can use this construct: https://github.com/lunarmodules/terminal.lua/blob/b908f8b2770b23e56fbdb340d189ffe723cfd884/src/terminal/scroll/stack.lua#L83-L86

The test-side of things being here: https://github.com/lunarmodules/terminal.lua/blob/b908f8b2770b23e56fbdb340d189ffe723cfd884/spec/05-scroll_stack_spec.lua#L5-L19